src/visitor.rs

```diff
@@ -641,8 +659,10 @@ impl Visitor {
         }
         // Crates like `pin_project` do some shenanigans to create and reference types that don't end up
         // in the doc index, but that should only happen within the root crate.
-        else if !id.0.starts_with(&format!("{}:", self.root_crate_id)) {
-            unreachable!("A type is referencing another type that is not in the index, and that type is from another crate.");
+        else if self.in_external_crate(id) {
+            unreachable!(
+                "An item referenced another item from an external crate that is not in the index."
+            );
         }
```

Comment:
I still have to figure out how to handle the docs visitor hitting this.

Reply:
OK, the issue was that we were choking on `doc(hidden)` items. I added a new kind of warning for those.